lib/ansible/modules/cloud/amazon/aws_s3.py

```diff
@@ -904,13 +934,16 @@ def main():
             module.fail_json(msg="Bucket parameter is required.")
 
     # Support for listing a set of keys
-    if mode == 'list':
+    if mode == 'list' or mode == 'listextended':
         exists = bucket_check(module, s3, bucket)
 
         # If the bucket does not exist then bail out
         if not exists:
             module.fail_json(msg="Target bucket (%s) cannot be found" % bucket)
 
+        if mode == 'listextended':
+            LIST_EXTENDED = True
+
```

Comment:
Or just pass LIST_EXTENDED from here to list_keys and then on to paginated_list from here.